dev/com.ibm.ws.transaction.fat.util/src/com/ibm/ws/transaction/fat/util/FATUtils.java

```diff
@@ -276,11 +284,127 @@ public static void stopServers(String[] toleratedMsgs, LibertyServer... servers)
+        final String method = "stopServer";
+
+        assertNotNull("Attempted to stop a null server", server);
+        int attempt = 0;
+        int maxAttempts = 5;
+        Log.info(c, method, "Stopping " + server.getServerName());
+        do {
+            if (attempt++ > 0) {
+                Log.info(c, method, "Waiting 5 seconds after stop failure before making attempt " + attempt);
+                try {
+                    Thread.sleep(5000);
+                } catch (Exception e) {
+                    Log.error(c, method, e);
+                }
+            }
+
+            if (!server.isStarted()) {
+                Log.info(c, method,
+                         "Server " + server.getServerName() + " is not started. No need to stop it.");
+                break;
+            }
+
+            ProgramOutput po = null;
+            try {
+                po = server.stopServer(toleratedMsgs);
+            } catch (Exception e) {
+                Log.error(c, method, e, "Server stop attempt " + attempt + " failed with return code " + (po != null ? po.getReturnCode() : "<unavailable>"));
+            }
+
+            if (po != null) {
+                String s;
+                int rc = po.getReturnCode();
+
+                Log.info(c, method, "ReturnCode: " + rc);
+
+                if (rc == 0) {
+                    break;
+                } else {
+                    String pid = server.getPid();
+                    Log.info(c, method,
+                             "Non zero return code stopping server " + server.getServerName() + "." + ((pid != null ? "(pid:" + pid + ")" : "")));
+
+                    s = po.getStdout();
+
+                    if (s != null && !s.isEmpty())
+                        Log.info(c, method, "Stdout: " + s.trim());
+
+                    s = po.getStderr();
+
+                    if (s != null && !s.isEmpty())
+                        Log.info(c, method, "Stderr: " + s.trim());
+
+                    server.printProcessHoldingPort(server.getHttpDefaultPort());
+                }
+            }
+        } while (attempt < maxAttempts);
+
+        if (server.isStarted()) {
+            server.postStopServerArchive();
+            throw new Exception("Failed to stop " + server.getServerName() + " after " + attempt + " attempts");
+        }
+	}
+
+	public static void stopServers(boolean parallel, String[] toleratedMsgs, LibertyServer... servers) throws Exception {
+		ArrayList<Future<?>> futures = new ArrayList<Future<?>>();
+		ExecutorService es;
+
+        if (parallel && servers.length > 1) {
+    		es = Executors.newFixedThreadPool(servers.length);
+
+    		for (LibertyServer server : servers) {
+    			futures.add(es.submit(new ServerStopper(toleratedMsgs, server)));
+    		}
+
+    		futures.parallelStream().forEach(future -> {
+                    try {
+						future.get();
+					} catch (Exception e) {
+						// Already logged
+					}
```

Comment:
How about:
```
Arrays.stream(servers).parallel().forEach(s -> {
    try {
        stopServer(toleratedMsgs, s);
    catch (Exception ignored) {}
});
```
For added bonus, incorporate the try/catch logic from run() within the stopServer() method, so that it doesn't need to declare "throws Exception", to make code in the forEach() even tidier.
Then there's no need for the ServerStopper class at all.

In fact, going further, you could then write this method like so:
```
public static void stopServers(boolean parallel, String[] toleratedMsgs, LibertyServer... servers) {
    Stream<LibertyServer> serverStream = Arrays.stream(servers);
    if (parallel) serverStream = serverStream.parallel();
    serverStream.forEach(s -> stopServer(toleratedMsgs, s));
}
```

Job done!